src/services/sig.py

```
@@ -423,8 +430,43 @@ async def executive_leave_sig(
+        sig = self.sig_repository.get_by_id(sig_id)
+        if sig is None:
+            raise HTTPException(404, detail=f"시그({sig_id=})가 존재하지 않습니다")
+        if (
+            executor.role < get_user_role_level("executive")
+            and sig.owner != executor.id
+        ):
+            raise HTTPException(403, detail="타인의 시그에 태그를 추가할 수 없습니다")
+        try:
+            sig_tag = self.sig_tag_repository.create(SIGTag(sig_id=sig_id, label=label))
+        except IntegrityError:
+            raise HTTPException(409, detail="이미 추가된 태그입니다")
```

Ruff (0.15.4)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)